### DISCUSSION TITLE
How to use gstreamer on windows?

### DISCUSSION THREAD

**Discussion by vipcxj** [microsoft/vcpkg#36609] [Q&A]
here is my CMakeLists.txt
``` CMakeLists.txt
...
add_executable(main ${CMAKE_CURRENT_SOURCE_DIR}/main.c)

find_package(PkgConfig REQUIRED)
pkg_check_modules(GST REQUIRED 
    gstreamer-1.0
    gstreamer-allocators-1.0
    gstreamer-base-1.0
    gstreamer-app-1.0 
    gstreamer-audio-1.0 
    gstreamer-controller-1.0 
    gstreamer-net-1.0 
    gstreamer-pbutils-1.0 
    gstreamer-plugins-base-1.0 
    gstreamer-rtp-1.0 
    gstreamer-sdp-1.0 
    gstreamer-tag-1.0 
    gstreamer-video-1.0
)
message("includes: ${GST_INCLUDE_DIRS}")
message("libraries: ${GST_LINK_LIBRARIES}")
message("library dir: ${GST_LIBRARY_DIRS}")
message("ldflags: ${GST_LDFLAGS}")
message("ldflags_other: ${GST_LDFLAGS_OTHER}")
message("cflags: ${GST_CFLAGS}")
message("${GST_CFLAGS_OTHER}")
target_include_directories(main PRIVATE "${GST_INCLUDE_DIRS}")
target_link_libraries(main PRIVATE PkgConfig::gstreamer ${GST_LINK_LIBRARIES})
target_compile_options(main PRIVATE ${GST_CFLAGS_OTHER})
```
my app compiled successfully. but element created failed. because lots of dll missing.
here is my dir tree of the output of build:
```
Release
|    ffi-8.dll
|    glib-2.0-0.dll
|    gmodule-2.0-0.dll
|    gobject-2.0-0.dll
|    gstreamer-1.0-0.dll
|    iconv-2.dll
|    intl-8.dll
|    main.exe
|    pcre2-8.dll
```

My app can't find any gstreamer plugins. So I configure gst_plugins manually to `${myprojectpath}\build\vcpkg_installed\x64-windows\plugins\gstreamer`, the gstreamer can find the plugins, but it failed to load them because missing dependent dll.

My quest is:
1 Do I have to configure gst_plugins manually? The plugins path is depend on port triplet, I don't want to hardcode it.
2 How can I tell cmake to automatically copy all needed dll to the output of the build? The same reason as above, I don't want to hardcode the path which depend on port triplet

**Comment by eladm-ultrawis** [comment]
@vipcxj did you find an answer? I am struggling with this as well

**Reply by vipcxj** [reply]
If triplet is x64-linux, gstreamer is compiled statically, and the current version has problems with static compilation, the plugin doesn't load automatically. I submitted this [pr](https://github.com/microsoft/vcpkg/pull/37392), updated to the latest 2.4.0, which supports full static compilation and can automatically load statically compiled plugins. If the triplet is x64-linux-dynamic, then it is dynamically compiled. vcpkg doesn't find the plugins automatically, but you can help gstreamer find them by configuring the GST_PLUGIN_PATH environment variable.
